Changing deploy to use GetOutputFilePath target

swjain23 opened the pull request:
Now instead of searching for the output file we are using the build target to get the output file path to deploy.

swjain23 committed:
Changing deploy to use GetOutputFilePath target

Now instead of searching for the output file we are using the build target to get the output file path to deploy.

nahk-ivanov commented on src/dotnet.ts:
```diff
@@ -103,6 +104,37 @@ export async function runDotNetCommand(dotNetSdk: DotNetInfo, args: string[], co
+	});
+}
+
+export async function runDotNetGetOutputFilePath(args: string[]): Promise<string> {
```
I don't think we need two methods here, the code is almost identical. Why won't you just pass a delegate for ```handleData```?
This function makes very little sense to be honest, as it tries to encapsulate the behavior (as name implies GetOutputFilePath), but it doesn't even restrict the build target name. I think you should have one generic function that takes one or two delegates, then you can have two exported wrapper functions that lock down the actual target that is being called and the way they process the output.

nahk-ivanov replied:
Or you keep one generic function here and do all customization in ```projectHelper.ts```

swjain23 committed:
Changes to not load tools service from release and deploy changes

swjain23 committed:
Deploy changes

Adding changes to execute language server commands on deploy. Handling notifications to print them out on output channel

nahk-ivanov commented on src/commands.ts:
```diff
@@ -87,26 +87,33 @@ async function deployCurrentProject(args, commandObserver: CommandObserver) {
+						let projectFileText = fs.readFileSync(filePath, 'utf8');
+						azdata.queryeditor.connect(filePath, connection.connectionId).then(() => {
+							commandObserver.logToOutputChannel(localize('extension.deploymentStartedMessage', '\nDeployment started {0}.', new Date().toLocaleString()));
+							commandObserver._client.sendRequest("query/executeDeployString", {owner_uri: filePath, query:projectFileText})
```
I'd probably call the command ```deployScript``` or something.

nahk-ivanov replied:
Or maybe ```deployString``` to be consistent, e.g. we change ```execute``` to ```deploy```...

swjain23 replied:
Hmmmmm, I added the command to execute_request in pgtoolsservice, so I was trying to follow the pattern.

nahk-ivanov replied:
Probably rename it there too? I agree we should be consistent...
I guess what confuses me now is ```executeString``` should be ```executeExecuteString``` if command itself is called "execute string". With current naming convention, if command is called "deploy string" then it should say just that, me thinks.

nahk-ivanov replied:
Hm... although looking on the other side they call them ```'EXECUTE_DOCUMENT_SELECTION_REQUEST', 'EXECUTE_STRING_REQUEST'```, which appear sot be that "execute request" is a nomenclature they use. That makes just "string" a request name, which is weird.
With this, maybe we can call it ```executeDeploy``` and on the python side call it ```EXECUTE_DEPLOY_REQUEST```? Would that make sense?

swjain23 replied:
Yeah that works. More consistent with what they have and still makes sense.

nahk-ivanov commented on src/commands.ts:
```diff
@@ -87,26 +87,33 @@ async function deployCurrentProject(args, commandObserver: CommandObserver) {
+					var connection = await azdata.connection.openConnectionDialog([Constants.providerId]);
+					if (connection) {
+						let projectFileText = fs.readFileSync(filePath, 'utf8');
+						azdata.queryeditor.connect(filePath, connection.connectionId).then(() => {
```
So this won't open any new windows rights? Do we need to disconnect it after deploy is done? Typically they would disconnect by closing the window and without the visual - when do we close the connection?

swjain23 replied:
Yes, this would not open any new windows. Let me see if need to disconnect it explicitly

nahk-ivanov commented on src/commands.ts:
```diff
@@ -87,26 +87,33 @@ async function deployCurrentProject(args, commandObserver: CommandObserver) {
+							commandObserver.logToOutputChannel(localize('extension.deploymentStartedMessage', '\nDeployment started {0}.', new Date().toLocaleString()));
+							commandObserver._client.sendRequest("query/executeDeployString", {owner_uri: filePath, query:projectFileText})
+							.then(() => { }, err => {
+								commandObserver.logToOutputChannel(localize('extension.FailedDeploy', 'Deployment failed with error: {0}', err.message));
```
Would we get multiple errors? If it still breaks it down to different statements and executes one at a time, then we would get multiple errors, right?

swjain23 replied:
This error is only when something is wrong with the request. The current execute implementation is such that if there is an error it would stop executing any further queries.

nahk-ivanov replied:
Ok, the execute sounds reasonable - if we fail to create a table, there is no reason to attempt to create objects that depend on it. Although if error is "table already exists", then we might want to just continue. We can do it later though, I don't think it's super critical and we need to better understand "update" scenario, when some objects already present.

nahk-ivanov commented on src/main.ts:
```diff
@@ -55,11 +56,15 @@ export async function activate(context: vscode.ExtensionContext) {
 		const installationComplete = Date.now();
 		let serverOptions = generateServerOptions(e);
 		languageClient = new SqlOpsDataClient(Constants.serviceName, serverOptions, clientOptions);
+		commandObserver._client = languageClient;
```
What do we need this for? Command observer doesn't look like something that would need language client. Also underscore usually indicates private member, you should be just randomly adding ```_``` members to classes whenever you want to...

swjain23 replied:
I don't think I need this anymore, will pass client as part of command and will get rid of this.

swjain23 committed:
Addressing PR Feedback

swjain23 committed:
Rename deploy request

swjain23 committed:
Update project template to use correct sdk nuget

swjain23 committed:
Rename CommandObserver.ts to commandObserver.ts

swjain23 committed:
Update filePatht o use uri as the client expects the file path
